### PR TITLE
Always Start in English Layout

### PR DESCRIPTION
**Pull request by onesounds**
## What's the PR
- Add settings that always start in English when you start flow.
  -  If the user is using a Native/English layout, Sometimes the input will start in an unwanted language.
- It always starts in English, regardless of your language status, so you can avoid typing incorrectly.
- This setting can be turned on and off; the default is off.

## Test Cases
- [x] Korean / Japanese / Chinese (pinyin)
- [x] When you turn the flow back on in the native keyboard layout state, typing must begin in English.

### CODE REVIEW

**Review by VictoriousRaptor — APPROVED** [review]

**Comment by onesounds** [comment]
> @onesounds @VictoriousRaptor when will users want to turn this option on?

It is necessary for a person who speaks a language other than English.
Since each language user uses both native language and English at the same time, there is an individual English state for each language. This language state is difficult to recognize input before it starts. So we need settings to start English state.Although it can start in the native language, it was judged that it is not general.

**Comment by onesounds** [comment]
https://github.com/Flow-Launcher/Flow.Launcher/issues/1640

This is why this request occurs. I judged that this function was better than that proposal. (I want to include that function, but that function is harder to implement than I thought. I have to import the ime32.dll to find out. current ime state)

This function is difficult to explain to people who only speak English, and it is difficult to test. This is why other programs don't have this function. I believe that this function is very important and worth applying.

**Comment by jjw24** [comment]
So if I have Korean IME and this option on, I type in search it will start in english but will it also change my language IME to English as well ? For example afterwards I go to website will I be typing English IME of my native IME?

**Comment by onesounds** [comment]
> So if I have Korean IME and this option on, I type in search it will start in english but will it also change my language IME to English as well ? For example afterwards I go to website will I be typing English IME of my native IME?

This is the difficulty of this function. Suppose a user using English, Korean, and Japanese IME.
In this case, IME can be switched through win+space.  This pr does not change this IME. 

However, in the case of Korean or JAPANESE IME, there are two LAYOUTs. (LAYOUT is a sub-item of IME.) It is in NATIVE and English. This PR's goal is always use ENGLISH LAYOUT when flow start. 

The ime that was in use is not changed. Layout is reflected only in the input window of flow.
In other words, when I open the flow while writing a document in Chinese, it starts with an English input, and when I close the flow and return to the original process, it is in the Chinese input state.

**Comment by jjw24** [comment]
Thanks for the explanation, let me take this pr for a spin.

**Comment by VictoriousRaptor** [comment]
> @onesounds @VictoriousRaptor when will users want to turn this option on?

Users can now assure that IME is in English so no issue with typing action keywords.

**Review by VictoriousRaptor — APPROVED** [review]

**Review by VictoriousRaptor — APPROVED** [review]

**Comment by sirito97** [comment]
I have tried this option and it is not working with Arabic, isn't it supposed to always search in English regardless of the selected language?

**Comment by VictoriousRaptor** [comment]
> I have tried this option and it is not working with Arabic, isn't it supposed to always search in English regardless of the selected language?

It's designed for CJK IME. IME is complex and not well documented. We will take a look.

**Comment by VictoriousRaptor** [comment]
@sirito97 How do you type Arabic? I just did some test and seems like Arabic input by Microsoft is not a IME but a keyboard layout, so our current method doesn't work. Sorry about that.

**Comment by sirito97** [comment]
> @sirito97 How do you type Arabic? I just did some test and seems like Arabic input by Microsoft is not a IME but a keyboard layout, so our current method doesn't work. Sorry about that.

@VictoriousRaptor Sorry for late reply, I really have no idea what is IME but here's a screenshot of my language input
Screenshot_20230124_072347

**Comment by VictoriousRaptor** [comment]
> > @sirito97 How do you type Arabic? I just did some test and seems like Arabic input by Microsoft is not a IME but a keyboard layout, so our current method doesn't work. Sorry about that.
> 
> @VictoriousRaptor Sorry for late reply, I really have no idea what is IME but here's a screenshot of my language input Screenshot_20230124_072347

it's difficult to explain what IME (Input Method Editor) is in a few words for people who don't use it. If you are curious please google it. May I ask you how do you switch between English and Arabic input? That can tell if you are using an IME.

**Comment by sirito97** [comment]
@VictoriousRaptor I switch by clicking left alt + shift here's a screenshot too
image

**Comment by sirito97** [comment]
> > @VictoriousRaptor I switch by clicking left alt + shift here's a screenshot too image
> 
> It's not an IME then. Sorry we are not able to fix it.

It's ok, I managed to fix it by ticking this option

**Comment by Sing303** [comment]
Not working for Russian language

1. Turned on "English Mode"
2. Switching to Russian language
3. Opened Flow Launcher. 

The behaviour is .
 - Language is still Russian

Expected behaviour
 - Regardless of the current layout, English characters are printed
